src/Ship.js

```diff
@@ -0,0 +1,45 @@
+class Ship {
+  constructor (itinerary) {
+    this.itinerary = itinerary;
+    this.currentPort = itinerary.ports[0];
+    this.previousPort = null;
+
+    this.currentPort.addShip(this);
+ 
+  }
```

Comment:
This could be bunched together without affecting readability
```
  constructor (itinerary) {
    this.itinerary = itinerary;
    this.currentPort = itinerary.ports[0];
    this.previousPort = null;
    this.currentPort.addShip(this);
  }
```